### PR TITLE
Feat: [Gaps-161]: Data Migration

### PR DESCRIPTION
**Pull request by Bernardvdv**
- Migration file added to migrate content from Url to Url Grouper

### CODE REVIEW

**Review by adam-murray — CHANGES_REQUESTED** [review]
We are missing the plugin migration at the moment. 
https://github.com/django-cms/djangocms-snippet/blob/aa450b38ba69ff96b30d52fef80f3f380e5e0f4a/djangocms_snippet/migrations/0011_cms4_plugin_data_migration.py#L5

File: djangocms_url_manager/migrations/0008_url_manager_data_migration.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    djangocms_versioning_config_enabled = UrlCMSAppConfig.djangocms_versioning_enabled
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    Url = apps.get_model("djangocms_url_manager", "Url")
+    UrlGrouper = apps.get_model("djangocms_url_manager", "UrlGrouper")
+    User = apps.get_model('auth', 'User')
+
+    url_contenttype = ContentType.objects.get(app_label='djangocms_url_manager', model='url')
+    url_queryset = Url.objects.all()
+
+    for url in url_queryset:
```
Comment (adam-murray) [inline djangocms_url_manager/migrations/0008_url_manager_data_migration.py:30]: I think we will need another migration for the UrlOverride as well.

File: djangocms_url_manager/migrations/0009_url_manager_data_migration.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from django.apps import apps as global_apps
+from django.contrib.contenttypes.management import create_contenttypes
+from django.db import migrations
+
+from djangocms_url_manager.cms_config import UrlCMSAppConfig
+from djangocms_url_manager.conf import (
+    DJANGOCMS_URL_MANAGER_VERSIONING_MIGRATION_USER_ID,
```
Comment (Aiky30) [inline djangocms_url_manager/migrations/0009_url_manager_data_migration.py:7]: This setting needs documenting in the documentation in this repo.

File: djangocms_url_manager/models.py [hunk 3]
```diff
@@ -185,10 +185,13 @@ class Meta:
...
-    internal_name = models.CharField(verbose_name=_("internal name"), max_length=120)
-    url = models.ForeignKey(
-        Url, verbose_name=_("url"), related_name="cms_plugins", on_delete=models.CASCADE
+    url_grouper = models.ForeignKey(
+        UrlGrouper,
+        on_delete=models.CASCADE,
+        related_name='%(app_label)s_%(class)s_url_grouper',
+        null=True
     )
+    internal_name = models.CharField(verbose_name=_("internal name"), max_length=120)
```
Comment (Aiky30) [inline djangocms_url_manager/models.py:194]:
```suggestion
```

We can remove all existence of the internal name on the plugin, it's a redundant field that was a mistake a long time ago. It has no use in the system. I've been waiting for the day when the plugin model was changed to edit this. It was a requirement for urls to allow you to find them in the system, they have no purpose on the plugin.

If you delete this field and the migration 11 file, rerun the makemigrations command and the deletions will be captured automatically.

File: djangocms_url_manager/migrations/0009_url_manager_data_migration.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+try:
+    from djangocms_versioning.constants import DRAFT
+
+    djangocms_versioning_installed = True
+except ImportError:
+    djangocms_versioning_installed = False
+
+
+def forwards(apps, schema_editor):
+    create_contenttypes(global_apps.get_app_config("djangocms_url_manager"))
```
Comment (Aiky30) [inline djangocms_url_manager/migrations/0009_url_manager_data_migration.py:19]:
```suggestion
```

You shouldn't need to create the content type, django takes care of this when loading up.

**Review by Aiky30 — CHANGES_REQUESTED** [review]

**Review by adam-murray — APPROVED** [review]

**Review by adam-murray — APPROVED** [review]

**Pull request closed without merging**